### PR TITLE
Xml aom fixes

### PR DESCRIPTION
**Pull request by pieterbos**
XML Marshalling of the AOM has a couple of problems:

- OPT and template gets serialized as archetype instead of operationaltemplate
- some map fields get marshalled as a default java map - should be StringDictionaryItem according to XSD
  - Archetype.otherMetaData
  - OperationalTemplate.componentTerminologies
  - OperationalTemplate.terminologyExtracts
- The JAXB context had a couple of classes added that could and should not be marshalled or unmarshalled. This is because a custom XML mapping exists for these types, that is automatically used instead. Things like ResourceDescription -> XMLResourceDescription. These non-xml custom classes have been removed from the JAXBContext obtained from JAXBUYtil.

This pull request fixes it. Note: this is a breaking change, as the XML format of the AOM (not the RM!) changes. However, it is asbolutely necessary, especially the root element is just broken, but also the map marshalling is a real problem. If anyone uses these particular fields in XML, they will need to change their code.

### CODE REVIEW

File: aom/src/main/java/com/nedap/archie/aom/Archetype.java [hunk 1]
```diff
@@ -74,10 +78,31 @@
...
+    //this is a specific map type to make a JAXB-adapter work. ugly jaxb
+    //alternative: define an extra field, use hooks to fill it just in time instead
+    @XmlTransient
+    private LinkedHashMap<String, String> otherMetaData = new LinkedHashMap<>();
 
     @XmlElement(name="other_meta_data")
-    //TODO: this probably requires a custom XmlAdapter
-    private Map<String, String> otherMetaData = new LinkedHashMap<>();
+    @JsonIgnore
+    private List<StringDictionaryItem> xmlOtherMetaData;
```
Comment (pieterbos) [inline aom/src/main/java/com/nedap/archie/aom/Archetype.java:88]: todo: make this field transient

File: openehr-rm/src/test/resources/com/nedap/archie/rm/composition/test_all_types.fixed.v1.xml [hunk 2]
```diff
@@ -4,15 +4,15 @@
         <value>Test all types</value>
     </name>
     <!-- the following is currently not supported by ARCHIE unmarshalling as of 14.7.2019 -->
```
Comment (pieterbos) [inline openehr-rm/src/test/resources/com/nedap/archie/rm/composition/test_all_types.fixed.v1.xml:6]: remove this line, as it is now supported.

File: aom/src/main/java/com/nedap/archie/aom/OperationalTemplate.java [hunk 3]
```diff
@@ -20,17 +30,78 @@
...
-@XmlRootElement(name="archetype")
+@XmlRootElement(name="operational_template")
 public class OperationalTemplate extends AuthoredArchetype {
 
     /**
      * terminology extracts from subarchetypes, for example snomed codes, multiple choice thingies, etc
      */
-    @XmlElement(name="terminology_extracts") //TODO: requires an adapter for JAXB to work
+    @XmlTransient
     private Map<String, ArchetypeTerminology> terminologyExtracts = new ConcurrentHashMap<>();//TODO: is this correct?
```
Comment (pieterbos) [inline aom/src/main/java/com/nedap/archie/aom/OperationalTemplate.java:40]: make this field transient

File: aom/src/main/java/com/nedap/archie/aom/OperationalTemplate.java [hunk 4]
```diff
@@ -20,17 +30,78 @@
...
-    @XmlElement(name="component_terminologies") //TODO: requires an adapter for JAXB to work
+    @XmlTransient
```
Comment (pieterbos) [inline aom/src/main/java/com/nedap/archie/aom/OperationalTemplate.java:41]: and this field transient as well

File: aom/src/main/java/com/nedap/archie/aom/OperationalTemplate.java [hunk 5]
```diff
@@ -20,17 +30,78 @@
...
     private Map<String, ArchetypeTerminology> componentTerminologies = new ConcurrentHashMap<>();
 
+    @XmlElement(name="terminology_extracts")
+    private List<XmlArchetypeTerminology> xmlTerminologyExtracts;
+    @XmlElement(name="component_terminologies")
+    private List<XmlArchetypeTerminology> xmlComponentTerminologies;
+
```
Comment (pieterbos) [inline aom/src/main/java/com/nedap/archie/aom/OperationalTemplate.java:48]: todo: mark these two fields @JsonIgnore and possibly transient

File: archie-utils/src/main/java/com/nedap/archie/xml/JAXBUtil.java [hunk 6]
```diff
@@ -60,10 +66,32 @@ public static synchronized JAXBContext createAOMContext() {
     public static synchronized JAXBContext createRMContext() {
         try {
             List<Class<?>> classes = Lists.newArrayList(ArchieRMInfoLookup.getInstance().getRmTypeNameToClassMap().values());
+            removeClasses(classes);
```
Comment (J3173) [inline archie-utils/src/main/java/com/nedap/archie/xml/JAXBUtil.java:69]: Shouldn't this call be in `createAOMContext` instead? Or both? And maybe `initArchieJaxbContext`?

File: aom/src/main/java/com/nedap/archie/aom/OperationalTemplate.java (hunk 5 above)
Comment (J3173) [reply aom/src/main/java/com/nedap/archie/aom/OperationalTemplate.java:48]: Jackson seems to already ignore these fields as there are no public getters.

**Review by J3173 — CHANGES_REQUESTED** [review]

**Review by J3173 — APPROVED** [review]

**Pull request merged**